File: javascript/matrices/2-TransposedMatrix.js
// Write a program that fills a 4x4 matrix with random values
// and displays the transposed matrix.

const matrix = [];
const transposedMatrix = [];

// fill atrix with random values
for (let i = 0; i < 4; i++) {
  matrix[i] = [];
  for (let j = 0; j < 4; j++) {
    matrix[i][j] = Math.floor(Math.random() * 10); // Random values from 0 to 9
  }
}

// Display the original matrix
console.log('The original matrix:');
console.log(matrix);

// Transpose the matrix
for (let i = 0; i < 4; i++) {
  transposedMatrix[i] = [];
  for (let j = 0; j < 4; j++) {
    transposedMatrix[i][j] = matrix[j][i];
  }
}

// Display the transposed matrix
console.log('\nThe transposed matrix:');
console.log(transposedMatrix);

File: python/matrices/2-TransposedMatrix.py
# Write a program that fills a 4x4 matrix with random values
# and displays the transposed matrix.

import random

# Initialize a 4x4 matrix
matrix = [[0, 0, 0, 0],
          [0, 0, 0, 0],
          [0, 0, 0, 0],
          [0, 0, 0, 0]]

# Fill the matrix random values 
for i in range(4):
  for j in range(4):
    matrix[i][j] = random.randint(1, 100)

# Display the original matrix
print("The original matrix:")
for row in matrix:
  print(row)

# Calculate the transpose of the matrix
transposed_matrix = [[0, 0, 0, 0],
                     [0, 0, 0, 0],
                     [0, 0, 0, 0],
                     [0, 0, 0, 0]]

for i in range(4):
  for j in range(4):
    transposed_matrix[i][j] = matrix[j][i]

# Display the transposed matrix
print("\nThe transposed matrix:")
for row in transposed_matrix:
  print(row)